seventeenth_hw/page_objects/search_page_pack/search_page.py

```diff
@@ -0,0 +1,62 @@
+from seventeenth_hw.page_objects.cart_page_pack.cart_page import CartPage
+from seventeenth_hw.page_objects.main_page_pack.main_page_locators import MainPageLocators
+from seventeenth_hw.page_objects.manager_application_form.manager_application_form import ManagerApplicationForm
+from seventeenth_hw.page_objects.search_page_pack.search_page_locators import SearchLocators
+from seventeenth_hw.utilities.config_reader import get_application_url
+from seventeenth_hw.utilities.web_ui.base_page import BasePage
+
+
+class SearchPage(BasePage):
+    def __init__(self, browser):
+        super().__init__(browser)
+        self.__main_page_locator = MainPageLocators()
+        self.__search_locator = SearchLocators()
+
+    def click_site_header_logo(self):
+        self._click(self.__main_page_locator.header_logo)
+        return self
+
+    def is_main_page_url_matches(self):
+        return self._wait_until_url_matches(get_application_url())
+
+    def click_manager_application_button(self):
+        self._click(self.__search_locator.manager_application_button)
+        return ManagerApplicationForm(self.browser)
+
+    def get_search_result_text(self):
+        return self._get_text(self.__search_locator.search_result_key)
+
+    def get_item_list_located(self):
+        return self._wait_until_all_elements_located(self.__search_locator.item_list)
```

Comment:
Looks like Base page function. The main idea of this approach is that all element findings are on BasePage, and on PageObjects you already work with elements